codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -139,11 +143,20 @@ const (
 	defaultQueueName           = "{{.type|allLowercasePlural}}"
 )
 
+type OptionsFn func (impl *{{.controllerImpl|raw}}) {{.controllerOptions|raw}}
```

Comment:
If you put this in `knative.dev/pkg/controller` than there would be a single `OptionsFn` type and we could provide constants that do things, if we wanted.